docs/developer-notes.md

```diff
@@ -139,6 +139,14 @@ CATcher uses the OAuth 2.0 protocol to authenticate users. Below is a summary of
 
 The authentication process is kicked off in the `AuthComponent`, but the code that co-ordinates steps 1 and 2 can be found in [`oauth.ts`](../oauth.ts)(For Electron) or `AuthService`(For Web). Step 2 requires a client secret granted to CATcher. To protect this, we run a web service, [gatekeeper](https://github.com/CATcher-org/gatekeeper) that executes step 2 on behalf of the client CATcher app.
 
+## Logging
+
+CATcher caches log data in `localStorage` within the browser. `localStorage` stores data as key, value pairs and behave similarly to cookies. The caching processes is handled the following way.
+
+- To ensure that all logs are routed through a custom error handling class, so that it can be manipulated and logged to the browser, we first let `ErrorHandlingService` implement the `ErrorHandler` class. (`ErrorHandler` is the default class responsible for logging errors, warnings and other data in Angular.)
+- Once the `ErrorHandler` class has been implemented and the `handleError` method is overriden in the `ErrorHandlingService`, we will replace the original `ErrorHandler` with our custom service in `app.module.ts` using a custom provider.
+- Since `LoggingService` is used to log all information passing through the `ErrorHandlingService` in the `handleError` method, we modify `LoggingService` such that all data is cached in `localStorage` and grouped with its respective `session`. (**Note**: A `session` signifies a new running instance of the application)
+
```

Comment:
Addtional point:

When adding your own logs, use the `LoggingService` instead of calling `console.log()`.

Reply:
Thats a good suggestion, but to new developers this can cause confusions especially when dealing with sections of the app where `LoggingService` is not injected into the component / service.

But yes, we should add additional logging to our processes so that if a bug occurs anywhere in the app we can check what the last successful action was. Maybe this can be made as an issue? What do you think? @dingyuchen @anubh-v

Reply:
> When adding your own logs, use the LoggingService instead of calling console.log().

I don't think we need to state this explicitly. The benefit of using `LoggingService` should be apparent since we indicate that it is meant for caching logs.